src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/AddNewUserTest.java

```diff
@@ -154,4 +156,37 @@ public void add_new_user_to_organisation_when_super_user_is_not_active_throws_40
         assertThat(newUserResponse).isNotNull();
         //assertThat((String) newUserResponse.get("message")).contains("Access Denied");
     }
+
+    @Test
+    public void add_new_user_with_caa_roles_to_organisation_should_return_201() {
+        List<String> userRoles = new ArrayList<>();
+        //userRoles.add("pui-caa");
+        //userRoles.add("caseworker-caa");
+        userRoles.add("pui-user-manager");
+        String firstName = "someName";
+        String lastName = "someLastName";
+        String email = randomAlphabetic(10) + "@hotmail.com".toLowerCase();
+
+        NewUserCreationRequest newUserCreationRequest = aNewUserCreationRequest()
+                .firstName(firstName)
+                .lastName(lastName)
+                .email(email)
+                .roles(userRoles)
+                .jurisdictions(createJurisdictions())
+                .build();
+
+        professionalApiClient.getMultipleAuthHeadersExternal(puiUserManager, firstName, lastName, email);
+
+        Map<String, Object> newUserResponse = professionalApiClient.addNewUserToAnOrganisation(orgIdentifierResponse, hmctsAdmin, newUserCreationRequest, HttpStatus.CREATED);
+        assertThat(newUserResponse).isNotNull();
+
+        Map<String, Object> searchUserResponse = professionalApiClient.searchUsersByOrganisation(orgIdentifierResponse, hmctsAdmin, "false", HttpStatus.OK);
+
+        List<Map> users = getNestedValue(searchUserResponse, "users");
+        Map newUserDetails = users.get(1);
+        List<String> superUserRoles = getNestedValue(newUserDetails, "roles");
+
+        //assertThat(superUserRoles).doesNotContain("pui-caa");
+        //assertThat(superUserRoles).doesNotContain("caseworker-caa");
+    }

```

Comment:
As we discussed this should have code to check if the roles are disabled or not.. if roles are disabled it should assert and exit from test case
If roles are enabled then it should validate the roles are added to super user

Reply:
This will be done in [RDCC-1388](https://tools.hmcts.net/jira/browse/RDCC-1388)